### PR TITLE
[DOC] Documentation for builtin gridSearch function

### PR DESCRIPTION
**Pull request by escher-m**
I have added the screenshots of the result in the shared google doc.

### CODE REVIEW

**Comment by j143** [comment]
Hi @escher-m - thank you for working on this. 

1. We can paste the results here itself. Copy paste the output though.
2. Get familiar with markdown syntax in GitHub comments. (Hint: Google this line!)

:smile_cat:

**Comment by escher-m** [comment]
The Console output is very long. The following is a snippet of the main errors that came up

```
Caused by: org.apache.sysds.runtime.DMLRuntimeException: Function .defaultNS::lmPredict is not a builtin function.
Caused by: org.apache.sysds.runtime.DMLRuntimeException: ERROR: Runtime error in program block generated from statement block between lines 2 and 6 -- Error evaluating instruction: CP°extfunct°.defaultNS°m_gridSearch°7°2°X=_mVar67·MATRIX·FP64°y=_mVar69·MATRIX·FP64°train=lm·SCALAR·STRING·true°predict=lmPredict·SCALAR·STRING·true°params=_Var70·LIST·UNKNOWN·false°paramValues=_Var77·LIST·UNKNOWN·false°verbose=true·SCALAR·BOOLEAN·true°B°opt
	
Caused by: org.apache.sysds.runtime.DMLRuntimeException: error executing function .defaultNS::m_gridSearch
Caused by: org.apache.sysds.runtime.DMLRuntimeException: ERROR: Runtime error in function program block generated from function statement block between lines 22 and 74 -- Error evaluating function program block
Caused by: org.apache.sysds.runtime.DMLRuntimeException: PARFOR: Failed to execute loop in parallel.
Caused by: org.apache.sysds.runtime.DMLRuntimeException: PARFOR: Number of executed tasks does not match the number of created tasks: tasks 0/11, iters 0/75.
```

**Comment by escher-m** [comment]
In addition to the function code, I took reference from [test](https://github.com/apache/systemml/blob/master/src/test/scripts/functions/builtin/GridSearchLM.dml)

**Comment by j143** [comment]
Hi @escher-m, search for the commit related to `gridsearch` in this repository, in that commit you would find some test files in .dml

**Comment by j143** [comment]
Meanwhile:

Figure out `how to work with git from commandline`. (Google this line!)

File: dev/docs/builtins-reference.md [hunk 1]
```diff
@@ -503,3 +504,44 @@ numClasses = 5
...
+| Matrix[Double] | Parameter combination |
+| Frame[Unknown] | Best results model |
+
+### Example
+```r
+X = rand (rows = 50, cols = 10)
+y = X %*% rand(rows = ncol(X), cols = 1)
+params = list("reg", "tol", "maxi")
+paramRanges = list(10^seq(0,-4), 10^seq(-5,-9), 10^seq(1,3))
+[B, opt]= gridSearch(X=X, y=y, train="lm", predict="lmPredict", params=params, paramValues=paramRanges, verbose = TRUE)
```
Comment (j143) [inline dev/docs/builtins-reference.md:545]: Is this not easier to `python` script? :smile:

**Comment by escher-m** [comment]
> Hi @escher-m, search for the commit related to `gridsearch` in this repository, in that commit you would find some test files in .dml

Yes, I did that. I added the link to it in my last comment above.

**Pull request closed without merging**